stdlib/private/SwiftPrivateThreadExtras/ThreadBarriers.swift

```diff
@@ -12,7 +12,7 @@
 
 #if os(macOS) || os(iOS) || os(watchOS) || os(tvOS)
 import Darwin
-#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
+#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku) || os(WASI)
```

Comment:
I think that it might be better to just not build `SwiftPrivateThreadExtras` on WASI

Reply:
This would make `RaceTest` that uses `SwiftPrivateThreadExtras` disabled, requiring us to disable quite a few tests. Do you think that be worth it?

Reply:
+1 to disabling the whole module, if it leads to simpler code. Race tests would not work (and are pointless) without threads anyway.